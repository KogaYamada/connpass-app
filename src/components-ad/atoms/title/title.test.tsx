import React from 'react';
import Title from './title.component';
import { shallow,configure, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('title snapshot tests', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  beforeEach(() => {
    wrapper=shallow(<Title>タイトル</Title>);
  })

  it('<Title>タイトル</Title> :propsなしのタイトル', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('<Title bold>タイトル</Title> :タイトルが太字', () => {
    wrapper.setProps({ bold: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('<Title sideBarColor="red">タイトル</Title> :タイトル横に赤いバー表示', () => {
    wrapper.setProps({ sideBarColor: 'red' });
    expect(wrapper).toMatchSnapshot();
  });

  it('<Title size="large">タイトル</Title> :サイズlarge', () => {
    wrapper.setProps({ size: 'large' });
    expect(wrapper).toMatchSnapshot();
  });

  it('<Title size="middle">タイトル</Title> :サイズmiddle', () => {
    wrapper.setProps({ size: 'middle' });
    expect(wrapper).toMatchSnapshot();
  });

  it('<Title size="small">タイトル</Title> :サイズsmall', () => {
    wrapper.setProps({ size: 'small' });
    expect(wrapper).toMatchSnapshot();
  });
});

it('title onPress',()=>{
  const onPress = jest.fn();
  const wrapper=shallow(<Title onPress={onPress}>タイトル</Title>);
  wrapper.dive().find('Text').simulate('press');
  expect(onPress).toHaveBeenCalled();
})
