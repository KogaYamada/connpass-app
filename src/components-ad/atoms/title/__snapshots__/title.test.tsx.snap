// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title snapshot tests <Title bold>タイトル</Title> :タイトルが太字 1`] = `
<View
  style={null}
>
  <Text
    style={
      Object {
        "fontSize": 36,
        "fontWeight": "bold",
      }
    }
  >
    タイトル
  </Text>
</View>
`;

exports[`title snapshot tests <Title sideBarColor="red">タイトル</Title> :タイトル横に赤いバー表示 1`] = `
<View
  style={
    Object {
      "borderLeftColor": "#A82402",
      "borderLeftWidth": 4,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 36,
      }
    }
  >
    タイトル
  </Text>
</View>
`;

exports[`title snapshot tests <Title size="large">タイトル</Title> :サイズlarge 1`] = `
<View
  style={null}
>
  <Text
    style={
      Object {
        "fontSize": 48,
      }
    }
  >
    タイトル
  </Text>
</View>
`;

exports[`title snapshot tests <Title size="middle">タイトル</Title> :サイズmiddle 1`] = `
<View
  style={null}
>
  <Text
    style={
      Object {
        "fontSize": 36,
      }
    }
  >
    タイトル
  </Text>
</View>
`;

exports[`title snapshot tests <Title size="small">タイトル</Title> :サイズsmall 1`] = `
<View
  style={null}
>
  <Text
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    タイトル
  </Text>
</View>
`;

exports[`title snapshot tests <Title>タイトル</Title> :propsなしのタイトル 1`] = `
<View
  style={null}
>
  <Text
    style={
      Object {
        "fontSize": 36,
      }
    }
  >
    タイトル
  </Text>
</View>
`;
